### PR TITLE
core-3713 Save column preferences for Data window in local storage.

### PR DESCRIPTION
**Pull request by sri2k1us**

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/DiskResourceView.java [hunk 1]
```diff
@@ -46,6 +48,10 @@
 
     void setWestPanelWidth(String width);
 
+    ColumnModel<DiskResource> getColumns();
+
+    void setColumnPreferences(Map<String, String> preferences);
+
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/DiskResourceView.java]: Please create Javadoc comments describing the purpose of each new public method.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/GridView.java [hunk 2]
```diff
@@ -267,4 +268,7 @@
     LiveGridCheckBoxSelectionModel getSelectionModel();
 
     void setSingleSelect();
+
+    void setColumnPreferences(Map<String, String> preferences);
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/GridView.java]: Please add javadoc comments describing purpose of new public method.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/views/grid/DiskResourceColumnModel.java [hunk 3]
```diff
@@ -49,6 +49,8 @@
                                                                                   CopyPathSelected.HasCopyPathSelectedEventHandlers {
 
 
+    private static List<ColumnConfig<DiskResource, ?>> list;
```
Comment (tedgin) [inline de-lib/src/main/java/org/iplantc/de/diskResource/client/views/grid/DiskResourceColumnModel.java]: Why did you a local variable of a static method get promoted to a private static field?  It's not being used anywhere else. Also, the original static method overwrites it each time, so it's not even acting like a cached copy.

Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/views/grid/DiskResourceColumnModel.java]: Good catch. I started with this approach and realized later i don't need get `ColumnConfig` from here. This change is not required.

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/GridView.java (hunk 2 above)
Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/GridView.java]: 👍

File: de-lib/src/main/java/org/iplantc/de/diskResource/client/DiskResourceView.java (hunk 1 above)
Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/diskResource/client/DiskResourceView.java]: 👍

**Comment by tedgin** [comment]
It now looks good to me.

**Pull request merged**